backend/modules/pdf_processing/ai_extraction_service.py

```
@@ -0,0 +1,592 @@
+                debug_log("[WARNING] No JSON array found in scenes response")
+                raise ValueError("Failed to extract scenes: No JSON array found in AI response")
+                
+        except Exception as e:
+            debug_log(f"[ERROR] Scene generation failed: {str(e)}")
+            raise
+    
+    async def generate_learning_outcomes(
+        self, 
+        combined_content: str, 
+        title: str, 
+        session_id: Optional[str] = None
```

Ruff (0.14.5)

Unused method argument: `session_id`

(ARG002)